mantidproject/mantid
# Avoid importing Qt when running "import mantid"

**Description of work.**

Some python algorithms import the reducer framework when they load. The reducer framework currently has a `mantidqt` dependency and forces the Qt libraries to load then the framework `mantid` module is loaded. This work removes that dependency by delaying the import until it is needed.

It also forces the PyQt api to use version 2 for all testing so that it is compatible with `qtpy` or else you might see the error "PyQt4.QtCore and PyQt5.QtCore modules both wrap the QObject class". This seems to be particularly problematic on macOS.

**Report to:** [nobody]. 

**To test:**

Check an interface such as "DGS Reduction" with the CNCS data in the system test data directory. Simply enter some sample data and click run - it should reduce successfully. Check this in both mantidplot and workbench.

*There is no associated issue.*

*This does not require release notes* because **it is an internal change.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
